### ISSUE TITLE
Some products not showing on Hungarian instance shopfront

### ISSUE THREAD

**Issue by pragai** [openfoodfoundation/openfoodnetwork#12751]
## Description
This error can be reproduced on the following page:

https://openfood.hu/anyahajoegyesulet/shop#/shop_panel

The products stop loading and the loader image is showing endlessly.

Checking the network tab of the developer toolbar a HTTP 422 error is shown with the following message:
exception	"uninitialized constant Spree::Image::MiniMagick\n\n rescue ActiveStorage::Error, MiniMagick::Error, ActionView::Template::Error => e\n ^^^^^^^^^^\nDid you mean? MiniMime"

## Expected Behavior
All products are shown.

## Actual Behaviour
Some products not shown, but instead the loader image shows up endlessly.

## Steps to Reproduce
1. Go to https://openfood.hu/anyahajoegyesulet/shop#/shop_panel
2. Scroll down until the loder image wont disaappear and products do not load.
3. It usually happens at product "Savanyú uborka "

## Animated Gif/Screenshot
![Screenshot from 2024-08-08 08-55-45](https://github.com/user-attachments/assets/2c23650e-2cec-4c5c-81b8-f0b0d194754a)

## Workaround
No workaround.

## Severity
bug-s1: a critical feature is broken: checkout, 

## Your Environment
Happens in all browsers, regardless of OS.

**Comment by mkllnk** [comment]
I couldn't reproduce the issue today because the shop is closed. But I found the Bugsnag report:

* https://app.bugsnag.com/yaycode/openfoodnetwork-hungary/errors/669f78ca9f9a7b06e8ee4cc5

My theory is that an error was raised processing a variant. And one of the referenced error types is `MiniMagick::Error`. But while we reference MiniMagick here, we don't actually load the code and the constant is undefined at this point.

You would think that the code should be automatically loaded by Rails but somehow it's not. And it maybe that it doesn't get loaded because variants are usually created in the background after an image upload. The Hungarian instance may just have some images that were not processed yet. But this is a general error that could happen to any instance, I believe.

I'll try to reproduce it locally. But for now I applied to hotfix on hu-prod:

```diff
diff --git a/config/application.rb b/config/application.rb
index 09d8fb5abe..37a32d599c 100644
--- a/config/application.rb
+++ b/config/application.rb
@@ -20,6 +20,9 @@ require "rails"
   end
 end
 
+# Hotfix for https://github.com/openfoodfoundation/openfoodnetwork/issues/12751
+require "mini_magick"
+
 require_relative "../lib/open_food_network/i18n_config"
 require_relative '../lib/spree/core/environment'
 require_relative '../lib/spree/core/mail_interceptor'
```

Let me know if that solves the problem and then we know that we are on the right path.
